src/find_sort_location.rs

```diff
@@ -0,0 +1,142 @@
+#![allow(dead_code)]
+
+use futures::future::join_all;
+
+use crate::{
+    field::Field,
+    helpers::ring::Ring,
+    replicated_secret_sharing::ReplicatedSecretSharing,
+    securemul::{ProtocolContext, SecureMul},
+};
+
+pub struct SecureSort<'a, R, F> {
+    context: &'a ProtocolContext<'a, R>,
+    share_of_one: ReplicatedSecretSharing<F>,
+}
+
+impl<F: Field, R: Ring> SecureSort<'_, R, F> {
+    pub async fn execute(
+        self,
+        shares: &[ReplicatedSecretSharing<F>],
```

Comment:
what is our strategy for dealing with associated data here? I assume we would have it as we care about sort stability

Reply:
We dont need to worry about the associated data within the sort protocol since the protocol will spit out a permutation which will be applied to the associated data.